### PR TITLE
Move perft to program

### LINKED ISSUE #40: Move perft to program
Doesn't need to be an XUnit test.
Call perft with the CLI, similar to benchmark `dotnet run -c Release --perft`

### PR DESCRIPTION
**Pull request by Sanjanarajbk**
Fixes #40

### CODE REVIEW

File: Rudim/Perft/PerftTest.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+            timer.Stop();
+
+            if(nodes != PerftDriver.Nodes){
+                Console.WriteLine($"There's a difference Expected: {nodes} Actual: {PerftDriver.Nodes}");
+                Environment.Exit(2);
+            }
+
+            History.ClearBoardHistory();
+            Console.WriteLine($"Execution Time: {timer.ElapsedMilliseconds} ms");
```
Comment (znxftw) [inline Rudim/Perft/PerftTest.cs:33]: Can you add some data on which test it is - before with XUnit we would be able to know which perft is taking how much.

"Execution Time: 100ms for Starting Position at depth 5"

You can check ToString() for BoardState already has some logic for common names so you can probably just to boardState.ToString() to print the position

File: Rudim/Perft/PerftTest.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+using Rudim.Board;
+using Rudim.Common;
+using System;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Rudim.Test.Perft
+{
+
+    public class PerftTest
```
Comment (znxftw) [inline Rudim/Perft/PerftTest.cs:11]: This can be static, so can the methods below

**Comment by Sanjanarajbk** [comment]
Modified to align with the suggestions.

File: .github/workflows/pipeline.yml [hunk 3]
```diff
@@ -33,7 +33,7 @@ jobs:
     - name: Build
       run: dotnet build --no-restore -c Release
     - name: Perft 
-      run: dotnet test --no-build --filter Category=Perft --verbosity normal -c Release
+      run: dotnet run -c Release --perft
     - name: Test
       run: dotnet test --no-build --filter Category!=Perft --verbosity normal -c Release --collect:"XPlat Code Coverage"
```
Comment (znxftw) [inline .github/workflows/pipeline.yml:38]: You can remove the `!=Perft` filter here now

**Pull request merged**